cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -187,9 +194,8 @@ public synchronized void delete(final Id.Namespace namespaceId) throws Exception
                                                                 namespaceId));
     }
 
-    // Namespace can be deleted. Revoke all privileges first
     authorizerInstantiator.get().enforce(namespace, SecurityRequestContext.toPrincipal(), Action.ADMIN);
-    authorizerInstantiator.get().revoke(namespace);
+    // Namespace can be deleted. Revoke all privileges first
```

Comment:
Yes.. this is the old comments. Should delete this. Privileges have to be revoked in the end since the deletion process also requires read access on namespace meta.